fusion-cli/build/plugins/instrumented-import-dependency-template-plugin.js

```diff
@@ -48,80 +99,107 @@ class InstrumentedImportDependencyTemplate extends ImportDependencyTemplate {
    */
   apply(dep /*: any */, source /*: any */, runtime /*: any */) {
     const depBlock = dep.block;
-    const content = runtime.moduleNamespacePromise({
-      block: dep.block,
-      module: dep.module,
-      request: dep.request,
-      strict: dep.originModule.buildMeta.strictHarmonyModule,
-      message: 'import()',
-    });
-
-    let chunkIds;
-
-    if (this.clientChunkIndex) {
+    let chunkIds = [];
+    let translationKeys = [];
+    if (dep instanceof InstrumentedImportDependency) {
+      const instrumentation = dep.getInstrumentation();
+      chunkIds = instrumentation.chunkIds;
+      translationKeys = instrumentation.translationKeys;
+    } else if (this.clientChunkIndex) {
+      // Template invoked without InstrumentedImportDependency
       // server-side, use values from client bundle
       let ids = this.clientChunkIndex.get(
         (dep.module && dep.module.resource) || dep.originModule.resource
       );
       chunkIds = ids ? Array.from(ids) : [];
     } else {
-      // client-side, use built-in values
-      chunkIds = getChunkGroupIds(depBlock.chunkGroup);
+      chunkIds = getChunkGroupIds(dep.block.chunkGroup);
     }
 
+    // This is a hack. Some production builds had undefined for the module.id
+    // which broke the build.
+    dep.module.id =
+      dep.module.id || (dep.module.identifier && dep.module.identifier());
```

Comment:
I tried a lot of different hooks to find one where `module.id` would always be defined (such as `afterOptimizeModuleIds`, but no luck. Without this line, the production build worked in some fixtures but failed in others. I couldn't find any reason why it wasn't defined consistently.